backend/battle/templates/battle/battle_form.html

```diff
@@ -13,8 +13,19 @@
         <p class="title_battle">Trainer</p>
         <p class="subtitle_battle">Choose your opponent!</p>
         <form method="POST" class="battle-form">{% csrf_token %}
-            {{ form.as_p }}
-                        
+            {{ form.non_field_errors }}
+            {% for hidden in form.hidden_fields %}
+                {{ hidden }}
+            {% endfor %}
+            <p>
+                <label>Opponent:</label>
+                <input id="opponent" name="opponent" type="text" list="users">
+                <datalist id="users" name="opponent">
+                    {% for results in users %}
```

Comment:
`for user` instead of `for results` is best here, because the plural implies multiple users.

Reply:
You're right, I'm going to fix this :)